challenge_5/C/karanchawla/src/challenge5.c

```diff
@@ -0,0 +1,109 @@
+/*
+Author: Karan Chawla
+Challenege #5
+Expected Running time O(k + n log n)
+where k is the position of the char in the new sorted string
+*/
+
+#include <stdio.h>
+#include <stdlib.h> 
+#include <time.h> 
+#include <string.h>
+
+
+//utility function used by qsort to sort the strings
+int comp (const void * elem1, const void * elem2) 
```

Comment:
http://www.cplusplus.com/reference/cstdlib/qsort/
`int compar (const void* p1, const void* p2);`
The function prototype used by qsort is like so^. Hence, I have to cast it from `void*` to `char*`. If I just define a method to compare two characters `qsort` doesn't work.